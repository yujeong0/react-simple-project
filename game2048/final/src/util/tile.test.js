// @ts-nocheck
import { moveTile } from './tile';

describe('moveTile', () => {
  function removeProps(tileList) {
    for (const tile of tileList) {
      delete tile.id;
      if (tile.isMerged === false) {
        delete tile.isMerged;
      }
      if (tile.isNew === false) {
        delete tile.isNew;
      }
    }
    return tileList;
  }
  describe('가로 8 8 4 4', () => {
    const tileList = [
      {
        id: 11,
        x: 1,
        y: 1,
        value: 8,
      },
      {
        id: 12,
        x: 2,
        y: 1,
        value: 8,
      },
      {
        id: 13,
        x: 3,
        y: 1,
        value: 4,
      },
      {
        id: 14,
        x: 4,
        y: 1,
        value: 4,
      },
    ];
    it('x=1', () => {
      expect(removeProps(moveTile({ x: 1, y: 0, tileList }))).toMatchSnapshot();
    });
    it('x=-1', () => {
      expect(
        removeProps(moveTile({ x: -1, y: 0, tileList })),
      ).toMatchSnapshot();
    });
    it('y=1', () => {
      expect(removeProps(moveTile({ x: 0, y: 1, tileList }))).toMatchSnapshot();
    });
    it('y=-1', () => {
      expect(
        removeProps(moveTile({ x: 0, y: -1, tileList })),
      ).toMatchSnapshot();
    });
  });
  describe('세로 8 8 4 4', () => {
    const tileList = [
      {
        id: 11,
        x: 1,
        y: 1,
        value: 8,
      },
      {
        id: 12,
        x: 1,
        y: 2,
        value: 8,
      },
      {
        id: 13,
        x: 1,
        y: 3,
        value: 4,
      },
      {
        id: 14,
        x: 1,
        y: 4,
        value: 4,
      },
    ];
    it('x=1', () => {
      expect(removeProps(moveTile({ x: 1, y: 0, tileList }))).toMatchSnapshot();
    });
    it('x=-1', () => {
      expect(
        removeProps(moveTile({ x: -1, y: 0, tileList })),
      ).toMatchSnapshot();
    });
    it('y=1', () => {
      expect(removeProps(moveTile({ x: 0, y: 1, tileList }))).toMatchSnapshot();
    });
    it('y=-1', () => {
      expect(
        removeProps(moveTile({ x: 0, y: -1, tileList })),
      ).toMatchSnapshot();
    });
  });
  describe('가로 2 4 8 8', () => {
    const tileList = [
      {
        id: 11,
        x: 1,
        y: 1,
        value: 2,
      },
      {
        id: 12,
        x: 2,
        y: 1,
        value: 4,
      },
      {
        id: 13,
        x: 3,
        y: 1,
        value: 8,
      },
      {
        id: 14,
        x: 4,
        y: 1,
        value: 8,
      },
    ];
    it('x=1', () => {
      expect(removeProps(moveTile({ x: 1, y: 0, tileList }))).toMatchSnapshot();
    });
    it('x=-1', () => {
      expect(
        removeProps(moveTile({ x: -1, y: 0, tileList })),
      ).toMatchSnapshot();
    });
  });
  describe('가로 2 x 8 8', () => {
    const tileList = [
      {
        id: 11,
        x: 1,
        y: 1,
        value: 2,
      },
      {
        id: 13,
        x: 3,
        y: 1,
        value: 8,
      },
      {
        id: 14,
        x: 4,
        y: 1,
        value: 8,
      },
    ];
    it('x=1', () => {
      expect(removeProps(moveTile({ x: 1, y: 0, tileList }))).toMatchSnapshot();
    });
    it('x=-1', () => {
      expect(
        removeProps(moveTile({ x: -1, y: 0, tileList })),
      ).toMatchSnapshot();
    });
  });
  describe('세로 2 x 8 8', () => {
    const tileList = [
      {
        id: 11,
        x: 1,
        y: 1,
        value: 2,
      },
      {
        id: 13,
        x: 1,
        y: 3,
        value: 8,
      },
      {
        id: 14,
        x: 1,
        y: 4,
        value: 8,
      },
    ];
    it('y=1', () => {
      expect(removeProps(moveTile({ x: 0, y: 1, tileList }))).toMatchSnapshot();
    });
    it('y=-1', () => {
      expect(
        removeProps(moveTile({ x: 0, y: -1, tileList })),
      ).toMatchSnapshot();
    });
  });
  describe('가로 여러 줄, 2 2 4 4, 8 x 4 4', () => {
    const tileList = [
      {
        id: 11,
        x: 1,
        y: 1,
        value: 2,
      },
      {
        id: 12,
        x: 2,
        y: 1,
        value: 2,
      },
      {
        id: 13,
        x: 3,
        y: 1,
        value: 4,
      },
      {
        id: 14,
        x: 4,
        y: 1,
        value: 4,
      },
      {
        id: 15,
        x: 1,
        y: 2,
        value: 8,
      },
      {
        id: 16,
        x: 3,
        y: 2,
        value: 4,
      },
      {
        id: 17,
        x: 4,
        y: 2,
        value: 4,
      },
    ];
    it('x=1', () => {
      expect(removeProps(moveTile({ x: 1, y: 0, tileList }))).toMatchSnapshot();
    });
    it('x=-1', () => {
      expect(
        removeProps(moveTile({ x: -1, y: 0, tileList })),
      ).toMatchSnapshot();
    });
  });
});
