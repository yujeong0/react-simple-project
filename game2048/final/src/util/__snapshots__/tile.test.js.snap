// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`moveTile 가로 2 4 8 8 x=-1 1`] = `
Array [
  Object {
    "value": 2,
    "x": 1,
    "y": 1,
  },
  Object {
    "value": 4,
    "x": 2,
    "y": 1,
  },
  Object {
    "isDisabled": true,
    "value": 8,
    "x": 3,
    "y": 1,
  },
  Object {
    "isDisabled": true,
    "value": 8,
    "x": 3,
    "y": 1,
  },
  Object {
    "isMerged": true,
    "value": 16,
    "x": 3,
    "y": 1,
  },
]
`;

exports[`moveTile 가로 2 4 8 8 x=1 1`] = `
Array [
  Object {
    "isDisabled": true,
    "value": 8,
    "x": 4,
    "y": 1,
  },
  Object {
    "isDisabled": true,
    "value": 8,
    "x": 4,
    "y": 1,
  },
  Object {
    "value": 4,
    "x": 3,
    "y": 1,
  },
  Object {
    "value": 2,
    "x": 2,
    "y": 1,
  },
  Object {
    "isMerged": true,
    "value": 16,
    "x": 4,
    "y": 1,
  },
]
`;

exports[`moveTile 가로 2 x 8 8 x=-1 1`] = `
Array [
  Object {
    "value": 2,
    "x": 1,
    "y": 1,
  },
  Object {
    "isDisabled": true,
    "value": 8,
    "x": 2,
    "y": 1,
  },
  Object {
    "isDisabled": true,
    "value": 8,
    "x": 2,
    "y": 1,
  },
  Object {
    "isMerged": true,
    "value": 16,
    "x": 2,
    "y": 1,
  },
]
`;

exports[`moveTile 가로 2 x 8 8 x=1 1`] = `
Array [
  Object {
    "isDisabled": true,
    "value": 8,
    "x": 4,
    "y": 1,
  },
  Object {
    "isDisabled": true,
    "value": 8,
    "x": 4,
    "y": 1,
  },
  Object {
    "value": 2,
    "x": 3,
    "y": 1,
  },
  Object {
    "isMerged": true,
    "value": 16,
    "x": 4,
    "y": 1,
  },
]
`;

exports[`moveTile 가로 8 8 4 4 x=-1 1`] = `
Array [
  Object {
    "isDisabled": true,
    "value": 8,
    "x": 1,
    "y": 1,
  },
  Object {
    "isDisabled": true,
    "value": 8,
    "x": 1,
    "y": 1,
  },
  Object {
    "isDisabled": true,
    "value": 4,
    "x": 2,
    "y": 1,
  },
  Object {
    "isDisabled": true,
    "value": 4,
    "x": 2,
    "y": 1,
  },
  Object {
    "isMerged": true,
    "value": 16,
    "x": 1,
    "y": 1,
  },
  Object {
    "isMerged": true,
    "value": 8,
    "x": 2,
    "y": 1,
  },
]
`;

exports[`moveTile 가로 8 8 4 4 x=1 1`] = `
Array [
  Object {
    "isDisabled": true,
    "value": 4,
    "x": 4,
    "y": 1,
  },
  Object {
    "isDisabled": true,
    "value": 4,
    "x": 4,
    "y": 1,
  },
  Object {
    "isDisabled": true,
    "value": 8,
    "x": 3,
    "y": 1,
  },
  Object {
    "isDisabled": true,
    "value": 8,
    "x": 3,
    "y": 1,
  },
  Object {
    "isMerged": true,
    "value": 8,
    "x": 4,
    "y": 1,
  },
  Object {
    "isMerged": true,
    "value": 16,
    "x": 3,
    "y": 1,
  },
]
`;

exports[`moveTile 가로 8 8 4 4 y=-1 1`] = `
Array [
  Object {
    "value": 8,
    "x": 1,
    "y": 1,
  },
  Object {
    "value": 8,
    "x": 2,
    "y": 1,
  },
  Object {
    "value": 4,
    "x": 3,
    "y": 1,
  },
  Object {
    "value": 4,
    "x": 4,
    "y": 1,
  },
]
`;

exports[`moveTile 가로 8 8 4 4 y=1 1`] = `
Array [
  Object {
    "value": 8,
    "x": 1,
    "y": 4,
  },
  Object {
    "value": 8,
    "x": 2,
    "y": 4,
  },
  Object {
    "value": 4,
    "x": 3,
    "y": 4,
  },
  Object {
    "value": 4,
    "x": 4,
    "y": 4,
  },
]
`;

exports[`moveTile 가로 여러 줄, 2 2 4 4, 8 x 4 4 x=-1 1`] = `
Array [
  Object {
    "isDisabled": true,
    "value": 2,
    "x": 1,
    "y": 1,
  },
  Object {
    "isDisabled": true,
    "value": 2,
    "x": 1,
    "y": 1,
  },
  Object {
    "isDisabled": true,
    "value": 4,
    "x": 2,
    "y": 1,
  },
  Object {
    "isDisabled": true,
    "value": 4,
    "x": 2,
    "y": 1,
  },
  Object {
    "value": 8,
    "x": 1,
    "y": 2,
  },
  Object {
    "isDisabled": true,
    "value": 4,
    "x": 2,
    "y": 2,
  },
  Object {
    "isDisabled": true,
    "value": 4,
    "x": 2,
    "y": 2,
  },
  Object {
    "isMerged": true,
    "value": 4,
    "x": 1,
    "y": 1,
  },
  Object {
    "isMerged": true,
    "value": 8,
    "x": 2,
    "y": 1,
  },
  Object {
    "isMerged": true,
    "value": 8,
    "x": 2,
    "y": 2,
  },
]
`;

exports[`moveTile 가로 여러 줄, 2 2 4 4, 8 x 4 4 x=1 1`] = `
Array [
  Object {
    "isDisabled": true,
    "value": 4,
    "x": 4,
    "y": 1,
  },
  Object {
    "isDisabled": true,
    "value": 4,
    "x": 4,
    "y": 1,
  },
  Object {
    "isDisabled": true,
    "value": 2,
    "x": 3,
    "y": 1,
  },
  Object {
    "isDisabled": true,
    "value": 2,
    "x": 3,
    "y": 1,
  },
  Object {
    "isDisabled": true,
    "value": 4,
    "x": 4,
    "y": 2,
  },
  Object {
    "isDisabled": true,
    "value": 4,
    "x": 4,
    "y": 2,
  },
  Object {
    "value": 8,
    "x": 3,
    "y": 2,
  },
  Object {
    "isMerged": true,
    "value": 8,
    "x": 4,
    "y": 1,
  },
  Object {
    "isMerged": true,
    "value": 4,
    "x": 3,
    "y": 1,
  },
  Object {
    "isMerged": true,
    "value": 8,
    "x": 4,
    "y": 2,
  },
]
`;

exports[`moveTile 세로 2 x 8 8 y=-1 1`] = `
Array [
  Object {
    "value": 2,
    "x": 1,
    "y": 1,
  },
  Object {
    "isDisabled": true,
    "value": 8,
    "x": 1,
    "y": 2,
  },
  Object {
    "isDisabled": true,
    "value": 8,
    "x": 1,
    "y": 2,
  },
  Object {
    "isMerged": true,
    "value": 16,
    "x": 1,
    "y": 2,
  },
]
`;

exports[`moveTile 세로 2 x 8 8 y=1 1`] = `
Array [
  Object {
    "isDisabled": true,
    "value": 8,
    "x": 1,
    "y": 4,
  },
  Object {
    "isDisabled": true,
    "value": 8,
    "x": 1,
    "y": 4,
  },
  Object {
    "value": 2,
    "x": 1,
    "y": 3,
  },
  Object {
    "isMerged": true,
    "value": 16,
    "x": 1,
    "y": 4,
  },
]
`;

exports[`moveTile 세로 8 8 4 4 x=-1 1`] = `
Array [
  Object {
    "value": 8,
    "x": 1,
    "y": 1,
  },
  Object {
    "value": 8,
    "x": 1,
    "y": 2,
  },
  Object {
    "value": 4,
    "x": 1,
    "y": 3,
  },
  Object {
    "value": 4,
    "x": 1,
    "y": 4,
  },
]
`;

exports[`moveTile 세로 8 8 4 4 x=1 1`] = `
Array [
  Object {
    "value": 8,
    "x": 4,
    "y": 1,
  },
  Object {
    "value": 8,
    "x": 4,
    "y": 2,
  },
  Object {
    "value": 4,
    "x": 4,
    "y": 3,
  },
  Object {
    "value": 4,
    "x": 4,
    "y": 4,
  },
]
`;

exports[`moveTile 세로 8 8 4 4 y=-1 1`] = `
Array [
  Object {
    "isDisabled": true,
    "value": 8,
    "x": 1,
    "y": 1,
  },
  Object {
    "isDisabled": true,
    "value": 8,
    "x": 1,
    "y": 1,
  },
  Object {
    "isDisabled": true,
    "value": 4,
    "x": 1,
    "y": 2,
  },
  Object {
    "isDisabled": true,
    "value": 4,
    "x": 1,
    "y": 2,
  },
  Object {
    "isMerged": true,
    "value": 16,
    "x": 1,
    "y": 1,
  },
  Object {
    "isMerged": true,
    "value": 8,
    "x": 1,
    "y": 2,
  },
]
`;

exports[`moveTile 세로 8 8 4 4 y=1 1`] = `
Array [
  Object {
    "isDisabled": true,
    "value": 4,
    "x": 1,
    "y": 4,
  },
  Object {
    "isDisabled": true,
    "value": 4,
    "x": 1,
    "y": 4,
  },
  Object {
    "isDisabled": true,
    "value": 8,
    "x": 1,
    "y": 3,
  },
  Object {
    "isDisabled": true,
    "value": 8,
    "x": 1,
    "y": 3,
  },
  Object {
    "isMerged": true,
    "value": 8,
    "x": 1,
    "y": 4,
  },
  Object {
    "isMerged": true,
    "value": 16,
    "x": 1,
    "y": 3,
  },
]
`;
